filebeat/tests/integration/autodiscover_test.go

```diff
@@ -0,0 +1,255 @@
+	err := provider.Create(clusterName, cluster.CreateWithV1Alpha4Config(&v1alpha4.Cluster{}))
+	if err != nil {
+		t.Fatalf("could not create cluster: %s", err)
+	}
+
+	t.Cleanup(func() {
+		if err := provider.Delete(clusterName, ""); err != nil {
+			t.Logf("could not delete K8s cluster: %s", err)
+		}
+	})
+
+	time.Sleep(30 * time.Second)
+
+	kubeConfig, err := provider.KubeConfig(clusterName, false)
+	if err != nil {
+		t.Fatalf("could not get KubeConfig: %s", err)
+	}
+
+	kubeConfigPath := filepath.Join(tempDir, "kube-config")
+	if err := os.WriteFile(kubeConfigPath, []byte(kubeConfig), 0666); err != nil {
+		t.Fatalf("cannot write kube config file: %s", err)
+	}
+
+	config, err := clientcmd.RESTConfigFromKubeConfig([]byte(kubeConfig))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "flog-pod-" + uid,
+			Namespace: "default",
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "flog",
+					Image: "mingrammer/flog",
+					Args:  []string{"-s", "1", "-d", "1", "-l"},
+				},
+			},
+		},
+	}
+
+	pod, err = clientset.CoreV1().Pods("default").Create(t.Context(), pod, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("could not create pod: %s", err)
+	}
+
+	t.Cleanup(func() {
+		err := clientset.CoreV1().Pods("default").Delete(context.Background(), pod.Name, metav1.DeleteOptions{})
+		if err != nil {
+			t.Logf("could not remove pod: %s", err)
+		}
+	})
+
+	// Wait for pod to be running
+	for range 60 {
+		pod, err = clientset.CoreV1().Pods("default").Get(context.Background(), pod.Name, metav1.GetOptions{})
+		if err != nil {
+			// We can probably ignore it and just time out
+			t.Logf("could not get pod: %s", err)
+		}
+
+		if pod.Status.Phase == corev1.PodRunning && len(pod.Status.ContainerStatuses) > 0 {
+			containerID := pod.Status.ContainerStatuses[0].ContainerID
+			if containerID != "" {
+				// Remove the runtime prefix (e.g., "containerd://")
+				if idx := strings.Index(containerID, "://"); idx != -1 {
+					return kubeConfigPath, pod.Spec.NodeName, containerID[idx+3:]
+				}
+				return kubeConfigPath, pod.Spec.NodeName, containerID
+			}
+		}
+
+		time.Sleep(time.Second)
+	}
+
+	t.Fatalf("pod did not start within timeout")
+	return kubeConfigPath, pod.Spec.NodeName, ""
+}
+
+// startFlogDocker starts a `mingrammer/flog` that logs one line every
+// second. The container ID is returned and the container is stopped at the
+// end of the test. On error the test fails by calling t.Fatalf
+func startFlogDocker(t *testing.T) string {
+	ctx := t.Context()
+	img := "mingrammer/flog"
+	cli, err := docker.NewClient(client.DefaultDockerHost, nil, nil, logp.NewNopLogger())
+	if err != nil {
+		t.Fatalf("cannot create Docker client: %s", err)
+	}
+
+	// Pull the image first
+	reader, err := cli.ImagePull(ctx, img, image.PullOptions{})
```

Comment:
I assume this pulls from latest, which seems to be [5 years old](https://hub.docker.com/r/mingrammer/). Can't we use the 0.4.3 tag which seems to be the same age? Just to avoid surprises.

Reply:
> Also, make sure this works when running the test suite on linux/arm64.

The tests passed on CI and we test on Linux ARM there.

Reply:
Thanks, I’m aware the tag is just a detail at this point, but it helps prevent the unlikely case of the latest image being updated.